### PR TITLE
Add CHANGELOG entry for 1.8.0

### PR DESCRIPTION
**Pull request by imurchie**
Preparation for releasing 1.8.0, scheduled for April 6, 2018

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,44 @@
...
+* Fix handling of screen recording
+* Make sure intent is broadcast when file is uploaded
+* Add possibility to broadcast device logs through WebSocket
+* Chromedriver
+  * Bundle version 2.37
+  * Automatically choose compatible Chromedriver for Chrome version
+    * `chromedriverExecutableDir` capability to specify where Chromedrivers are located
+    * `chromedriverChromeMappingFile` capability to specify what Chromedriver version corresponds to what Chrome version
+* Fix Handling of idle state waiting
+* Speed up screenshot logic
```
Comment (mykola-mokhnach) [inline CHANGELOG.md]: Only for UIA2

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,44 @@
...
+* Add ability to record screen
+* Fix handling of custom SSL certificate on simulators
+* Better handling of `xcodebuild` processes
+* Maintain keychains while upgrading apps
+* Better handle simulators in unexpected states
+
+#### Android
+* Support for Android P (API Level 28)
+* Fix handling of webview contexts when resetting apps
+* Fix handling of screen recording
```
Comment (mykola-mokhnach) [inline CHANGELOG.md]: Screen recording is also available for iOS

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,44 @@
...
+===================================
+
+Appium 1.8.0 introduces full support for the [W3C WebDriver specification](https://www.w3.org/TR/webdriver/)
+([digested version of the spec](https://github.com/jlipps/simple-wd-spec)).
+
+#### General
+* Minimum [NodeJS]() version moved to 6
+* Add methods for
+  * getting and setting the clipboard contents
+  * get full screen screenshots
```
Comment (mykola-mokhnach) [inline CHANGELOG.md]: We've also fixed error codes handling and updated the webdriver error codes: 
https://github.com/appium/appium-base-driver/pull/192
https://github.com/appium/appium-base-driver/pull/193

**Comment by mykola-mokhnach** [comment]
Other interesting features:

https://github.com/appium/appium-base-driver/pull/187 - URLs caching

https://github.com/appium/appium-base-driver/pull/179 - Advanced applications management (both Android and iOS)

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,44 @@
...
+#### iOS
+* Support for iOS 11.3/Xcode 9.3
+* Fix handling of process arguments
+* Add capabilities
+  * `shutdownOtherSimulators` - shutdown other running simulators at session start up
+* Add ability to record screen
+* Fix handling of custom SSL certificate on simulators
+* Better handling of `xcodebuild` processes
+* Maintain keychains while upgrading apps
+* Better handle simulators in unexpected states
```
Comment (KazuCocoa) [inline CHANGELOG.md]: Can we address measuring performance feature as well?

- https://github.com/appium/appium-xcuitest-driver/pull/637
- https://github.com/appium/appium/blob/master/docs/en/writing-running-appium/ios/ios-xctest-performance.md

**Comment by mykola-mokhnach** [comment]
+ bugfix: `moveTo` coordinates are now always considered as absolute (iOS & Android)

**Comment by KazuCocoa** [comment]
Is it better to address a guide about `wait` in TouchAction? The guide is inserting `wait` when users try to `moveTo` action to emulate swipe.

One of my friends asked me and I investigated the difference touch action between 1.7.2 and 1.8.0.

https://github.com/Magic-Pod/AppiumRegressionCheck/issues/6#issuecomment-377914333

test code(JS)

```js
let action = new TouchAction(driver);
action.press({x:200, y:200}).moveTo({x:200, y:0}).release();
```

With 1.7.2, the code issues the below command. `dragfromtoforduration` is called.

```
2018-04-02 12:37:42:951 - info: [HTTP] --> POST /wd/hub/session/4969076e-df2b-43fb-ad77-302eda1a8c22/touch/perform {"actions":[{"action":"press","options":{"x":200,"y":200}},{"action":"moveTo","options":{"x":200,"y":0}},{"action":"release","options":{}}]}
2018-04-02 12:37:42:955 - info: [debug] [MJSONWP] Calling AppiumDriver.performTouch() with args: [[{"action":"press","options":{"x":200,"y":200}},{"action":"moveTo","options":{"x":200,"y":0}},{"action":"release","options":{}}],"4969076e-df2b-43fb-ad77-302eda1a8c22"]
2018-04-02 12:37:42:956 - info: [debug] [XCUITest] Executing command 'performTouch'
2018-04-02 12:37:42:961 - info: [debug] [XCUITest] Received the following touch action: press(options={"x":200,"y":200})-moveTo(options={"x":200,"y":0})-release(options={})
2018-04-02 12:37:42:961 - info: [debug] [XCUITest] Found matching gesture: scroll
2018-04-02 12:37:42:980 - info: [debug] [JSONWP Proxy] Proxying [POST /wda/dragfromtoforduration] to [POST http://localhost:8101/session/29D062D8-397B-43B0-9668-8F235C0E20E1/wda/dragfromtoforduration] with body: {"fromX":200,"fromY":200,"toX":400,"toY":200,"duration":0}
2018-04-02 12:37:44:202 - info: [debug] [JSONWP Proxy] Got response with status 200: {"value":{},"sessionId":"29D062D8-397B-43B0-9668-8F235C0E20E1","status":0}
2018-04-02 12:37:44:203 - info: [debug] [MJSONWP] Responding to client with driver.performTouch() result: null
2018-04-02 12:37:44:205 - info: [HTTP] <-- POST /wd/hub/session/4969076e-df2b-43fb-ad77-302eda1a8c22/touch/perform 200 1253 ms - 76 
```

But with 1.8.0, the command issues. `dragfromtoforduration` isn't called.

```
2018-04-02 12:46:35:629 - info: [HTTP] <-- POST /wd/hub/session/41d7a8b1-b464-4110-b9f7-4234403d57b2/touch/perform 200 241 ms - 74
2018-04-02 12:46:35:629 - info: [HTTP] 
2018-04-02 12:46:36:645 - info: [HTTP] --> POST /wd/hub/session/41d7a8b1-b464-4110-b9f7-4234403d57b2/touch/perform
2018-04-02 12:46:36:645 - info: [HTTP] {"actions":[{"action":"press","options":{"x":200,"y":200}},{"action":"moveTo","options":{"x":200,"y":0}},{"action":"release","options":{}}]}
2018-04-02 12:46:36:646 - info: [debug] [MJSONWP] Calling AppiumDriver.performTouch() with args: [[{"action":"press","options":{"x":200,"y":200}},{"action":"moveTo","options":{"x":200,"y":0}},{"action":"release","options":{}}],"41d7a8b1-b464-4110-b9f7-4234403d57b2"]
2018-04-02 12:46:36:646 - info: [debug] [XCUITest] Executing command 'performTouch'
2018-04-02 12:46:36:648 - info: [debug] [XCUITest] Received the following touch action: press(options={"x":200,"y":200})-moveTo(options={"x":200,"y":0})-release(options={})
2018-04-02 12:46:36:649 - info: [debug] [JSONWP Proxy] Proxying [POST /wda/touch/perform] to [POST http://localhost:8101/session/2D74DF11-F3C2-4176-98F8-616E4DA7585C/wda/touch/perform] with body: {"actions":[{"action":"press","options":{"x":200,"y":200}},{"action":"moveTo","options":{"x":200,"y":0}},{"action":"release","options":{}}]}
2018-04-02 12:46:36:847 - info: [debug] [JSONWP Proxy] Got response with status 200: {"value":{},"sessionId":"2D74DF11-F3C2-4176-98F8-616E4DA7585C","status":0}
2018-04-02 12:46:36:848 - info: [debug] [MJSONWP] Responding to client with driver.performTouch() result: {}
2018-04-02 12:46:36:849 - info: [HTTP] <-- POST /wd/hub/session/41d7a8b1-b464-4110-b9f7-4234403d57b2/touch/perform 200 203 ms - 74
```

The log has no issue but with the 1.8.0, we need to insert `wait` method like `action.press({x:200, y:200}).wait({ms: 500}).moveTo({x:200, y:1}).release();` to swipe on device accurately. I guess the command itself works from the log but animation doesn't happen on device. But once we insert `wait` in the command, the swipe animation also happened expectedly.

In the Ruby client, `swipe` action wrap `press, wait, moveTo` and the client should work fine both  1.7.2 and 1.8.0. But with the other client, I'm not sure they provide such wrapper and if they don't provide them, this issue probably happens on users' environment.

I hope my explanation is understandable...

**Comment by Bamieh** [comment]
One feature I am waiting for in this release is not mentioned here in the logs, `support for registering through https protocol in selenium grid`

**Comment by jlipps** [comment]
@NozomiIto / @KazuCocoa: without a `wait` action to denote the time, how can we express the duration for thes swipe? this is one reason the new w3c actions spec is better moving forward.

**Comment by KazuCocoa** [comment]
@jlipps 
Personally, inserting `wait` is good to clear how durations the action has.

I was afraded that 1.8.0 breaks user test suite without guiding the `wait` thing to uses. If appium users upgrade their environemnt to 1.8.0 without knowing that, some test will break. So, I thought preparing migration guide like the below.

- Users, who implement `swipe` action using `moveTo` like the below, should insert `wait` since our `moveTo` action doesn't include duration.

```js
# Before
let action = new TouchAction(driver);
action.press({x:200, y:200}).moveTo({x:200, y:0}).release();

# After
let action = new TouchAction(driver);
action.press({x:200, y:200}).wait({ms: 500}).moveTo({x:200, y:0}).release();
```

**Comment by mykola-mokhnach** [comment]
@KazuCocoa Previously xcuitest driver was trying to parse and match the actual actions chain to the one of available native calls. That is why the set of supported chains was pretty limited. Since 1.8.0 this is not the case anymore, since all chain item calls are "fair" and are mapped to native XCTest calls inside WDA.

**Comment by NozomiIto** [comment]
@mykola-mokhnach @jlipps 
Thanks for the reply!
Is this moveTo behavior change the expected correct behavior change?
If so, I will add `wait` to all my test codes so that it work well with 1.8.0.
(And as @KazuCocoa says, the necessity of the change for the existing test codes should be notified to users in README etc)

**Comment by dpgraham** [comment]
I think we're good to merge this now. Doing a release today.

If there's anything missing, just make another PR.

**Comment by KazuCocoa** [comment]
Yup, thanks!

**Pull request merged**